test/services/dumpers/agent-nodejs/agent-nodejs-dumper.test.ts

```diff
@@ -0,0 +1,482 @@
+  describe('dump', () => {
+    const configs = [
+      {
+        name: 'mongodb',
+        appConfig: {
+          applicationName: 'test-output/mongodb',
+          appHostname: 'localhost',
+          appPort: 1654,
+          path: appRoot,
+        },
+        dbConfig: {
+          dbDialect: 'mongodb',
+          dbConnectionUrl: 'mongodb://localhost:27016',
+          ssl: false,
+          dbSchema: 'public',
+        },
+        forestAuthSecret: 'forestAuthSecret',
+        forestEnvSecret: 'forestEnvSecret',
+      },
+      {
+        name: 'postgres',
+        appConfig: {
+          applicationName: 'test-output/postgres',
+          appHostname: 'localhost',
+          appPort: 1654,
+          path: appRoot,
+        },
+        dbConfig: {
+          dbDialect: 'postgres',
+          dbConnectionUrl: 'postgres://localhost:54369',
+          ssl: false,
+          dbSchema: 'public',
+        },
+        forestAuthSecret: 'forestAuthSecret',
+        forestEnvSecret: 'forestEnvSecret',
+      },
+      {
+        name: 'mysql',
+        appConfig: {
+          applicationName: 'test-output/mysql',
+          appHostname: 'localhost',
+          appPort: 1654,
+          path: appRoot,
+        },
+        dbConfig: {
+          dbDialect: 'mysql',
+          dbConnectionUrl: 'mysql://localhost:8999',
+          ssl: false,
+          dbSchema: 'public',
+        },
+      },
+      {
+        name: 'mariadb',
+        appConfig: {
+          applicationName: 'test-output/mariadb',
+          appHostname: 'localhost',
+          appPort: 1654,
+          path: appRoot,
+        },
+        dbConfig: {
+          dbDialect: 'mariadb',
+          dbConnectionUrl: 'mariadb://localhost:3305',
+          ssl: false,
+          dbSchema: 'public',
+        },
+      },
+      {
+        name: 'mssql',
+        appConfig: {
+          applicationName: 'test-output/mssql',
+          appHostname: 'localhost',
+          appPort: 1654,
+          path: appRoot,
+        },
+        dbConfig: {
+          dbDialect: 'mssql',
+          dbConnectionUrl: 'mssql://localhost:1432',
+          ssl: false,
+          dbSchema: 'public',
+        },
+      },
+    ];
+
+    const files = [
+      'index.js',
+      'package.json',
+      'Dockerfile',
+      'docker-compose.yml',
+      '.env',
+      '.gitignore',
+      '.dockerignore',
+    ];
+
+    configs.forEach(config => {
+      describe(`database ${config.name}`, () => {
+        async function dump() {
+          const injectedContext = Context.execute(defaultPlan);
+          const dumper = new AgentNodeJsDumper(injectedContext);
+          await dumper.dump(
+            {
+              appConfig: config.appConfig,
+              dbConfig: config.dbConfig,
+              forestAuthSecret: 'forestAuthSecret',
+              forestEnvSecret: 'forestEnvSecret',
+            },
+            {},
+          );
+        }
+
+        files.forEach(fileName => {
```

Comment:
Please it.each(), purposely made for that and not breaking ide jest integration

Reply:
👍